tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+        // In the implementation, essentially unacknowledged IBC packets.
+        outstandingPacketsToProvider: List[VSCMaturedPacket],
+    }
+
+    // utility function: returns a consumer state that is initialized minimally.
+    pure def GetEmptyConsumerState(): ConsumerState = 
+        {
+            chainState: GetEmptyChainState,
+            maturationTimes: Map(),
+            outstandingPacketsToProvider: List(),
+            receivedVSCPackets: List(),
+        }
+
+    // the state of the protocol consists of the composition of the state of one provider chain with potentially many consumer chains.
+    type ProtocolState = {
+        providerState: ProviderState,
+        // the state of each consumer chain.
+        // note that we assume that this contains all consumer chains that may ever exist,
+        // and consumer chains that are currently not running will have providerState.consumerStatus == UNUSED or STOPPED.
+        consumerStates: Chain -> ConsumerState
+    }
+
+    // gets a protocol state that is initialized minimally.
+    pure def GetEmptyProtocolState(): ProtocolState = 
+        {
+            providerState: GetEmptyProviderState,
+            consumerStates: Map(),
+        }
+
+    type Error = {
+        message: str
+    }
+
+    // we return either a result or an error.
+    // if hasError is true, newState may be arbitrary, but the error will be meaningful.
+    // if hasError is false, error may be arbitrary, but newState will be meaningful.
+    type Result = {
+        hasError: bool,
+        newState: ProtocolState,
+        error: Error
+    }
+
+    pure def Ok(newState: ProtocolState): Result = {
+        {
+            hasError: false,
+            newState: newState,
+            error: {
+                message: ""
+            }
+        }
+    }
+
+    pure def Err(msg: str): Result = {
+        {
+            hasError: true,
+            newState: {
+                providerState: {
+                    chainState: {
+                        votingPowerHistory: List(),
+                        currentValidatorSet: Map(),
+                        lastTimestamp: 0,
+                    },
+                    outstandingPacketsToConsumer: Map(),
+                    receivedMaturations: Set(),
+                    sentVSCPackets: Map(),
+                    providerValidatorSetChangedInThisBlock: false,
+                    consumerStatus: Map(),
+                    runningVscId: 0,
+                },
+                consumerStates: Map(),
+            },
+            error: {
+                message: msg
+            }
+        }
+    }
+
+    // possible consumer statuses
+    pure val STOPPED = "stopped" // the chain was once a consumer chain, but has been voluntarily dropped by the provider.
+    pure val TIMEDOUT = "timedout" // the chain has timed out and was dropped by the provider. This is only used for involuntary drops.
```

Comment:
Do we _win_ anything by making this distinction on whether a chain was dropped by the consumer or it was voluntarily stopped? It feels we could put both under the umbrella of stopped for simplicity?

If you leave this as is, then it might make sense to explain what "timed out" means here -- that the provider has not received a matured packet  from the chain the last `CcvTimeout` time since the packet was sent.

Reply:
I think it makes it more human-understandable.
It also allows me to write a sanity check that ensures that packets can also time out,
so it allows more fine-grained testing.

I think it makes the model more explicit and the traces easier to understand,
and I think the tradeoff of slightly more complexity is worth it